packages/components/tests/lib/controllers/app-controller.test.js

```diff
@@ -1,839 +1,1389 @@
+
+    return appController
+  }
+
+  beforeAll(() => {
+    Vue.config.productionTip = false // Disable a Vue production mode warning
+    Vue.config.devtools = false // Disable a Devtools extension warning
   })
+
+  beforeEach(() => {
+    uiState = BaseTestHelp.createUIState()
+  })
+
   afterEach(() => {
     jest.resetModules()
-    uiC = undefined
   })
+
   afterAll(() => {
     jest.clearAllMocks()
   })
 
-  let l10n = new L10n()
-    .addMessages(enUS, Locales.en_US)
-    .addMessages(enGB, Locales.en_GB)
-    .setLocale(Locales.en_US)
-
-  let latID = Constants.LANG_LATIN
-  let araID = Constants.LANG_ARABIC
-/*
-  it('1 AppController - create object with min arguments', async () => {
-    expect(uiC.options).toHaveProperty('uiTypePanel')
-    expect(uiC.options).toHaveProperty('uiTypePopup')
+  it('AppController - constructor: should create an instance with default arguments', () => {
+    appC = new AppController(uiState)
+    expect(appC).toBeInstanceOf(AppController)
+  })
 
-    expect(uiC.options.irregularBaseFontSizeClassName.length).toBeGreaterThan(0)
-    expect(uiC.irregularBaseFontSize).toBeDefined()
-    expect(uiC.options.app).toBeDefined()
+  it('AppController - constructor: should initialize all properties', () => {
+    appC = new AppController(uiState)
+    expect(appC.state).toBe(uiState)
+    expect(appC.api).toEqual({})
+    expect(appC.isInitialized).toBeFalsy()
+    expect(appC.isActivated).toBeFalsy()
+    expect(appC.isDeactivated).toBeFalsy()
+  })
 
-    expect(uiC.options.template).toHaveProperty('html')
-    expect(uiC.options.template).toHaveProperty('panelId')
+  it('AppController - constructor: should initialize internal options with the constructor arguments', () => {
+    const customOptions = {
+      clientId: 'alpheios-components',
+      app: {
+        name: 'Test App'
+      }
+    }
+    appC = new AppController(uiState, customOptions)
+    expect(appC._options).toMatchObject(customOptions)
+  })
 
-    expect(uiC.options.template).toHaveProperty('defaultPanelComponent')
-    expect(uiC.options.template).toHaveProperty('popupId')
+  it('AppController - create: should create an instance with default arguments', () => {
+    appC = AppController.jestCreate(uiState)
+    appC.registerModule(AuthModule)
+    expect(appC).toBeInstanceOf(AppController)
+  })
 
-    expect(uiC.options.template).toHaveProperty('defaultPopupComponent')
-    expect(uiC.options.template).toHaveProperty('draggable')
-    expect(uiC.options.template).toHaveProperty('resizable')
+  it('AppController - create: an instance created should have L10n and Lexis modules', () => {
+    appC = AppController.jestCreate(uiState)
+    expect(appC.hasModule('l10n')).toBeTruthy()
+    expect(appC.hasModule('lexis')).toBeTruthy()
+  })
 
-    expect(uiC.zIndex).toBeGreaterThan(0)
+  it('AppController - setOptions: should combine options provided and the default ones', () => {
```

Comment:
this test doesn't look any different than the previous one that tests options. Is it actually testing if options are combined?

Reply:
The purpose of this test is to check that the options supplied would be merged properly with the default options. So it creates a custom options object and checks whether the `AppController.setOptions()` will place it into the resulting output.

We can also add a check that without the `customOptions` object supplied the `AppController.setOptions()` output will not contain the custom options object.

Does it make sense?

Reply:
It makes sense, I just wasn't seeing how this test differed from the one on lines 110-119.  I.e. it didn't seem to be testing a merge of customOptions with default options, it just tested that what was supplied is what was set.